lib/trento/application/usecases/alerting/alerting.ex

```diff
@@ -15,51 +15,52 @@ defmodule Trento.Application.UseCases.Alerting do
 
   require Logger
 
-  @spec notify_host_heartbeating_failure(binary()) :: :ok | {:error, any}
+  @spec notify_host_heartbeating_failure(binary()) :: :ok
   def notify_host_heartbeating_failure(host_id) do
     %HostReadModel{hostname: hostname} = Trento.Repo.get!(HostReadModel, host_id)
 
     EmailAlert.alert("Host", "hostname", hostname, "heartbeat failed")
     |> deliver_notification()
   end
 
-  @spec notify_critical_cluster_health(binary()) :: :ok | {:error, any}
+  @spec notify_critical_cluster_health(binary()) :: :ok
   def notify_critical_cluster_health(cluster_id) do
     %ClusterReadModel{name: name} = Trento.Repo.get!(ClusterReadModel, cluster_id)
 
     EmailAlert.alert("Cluster", "name", name, "health is now in critical state")
     |> deliver_notification()
   end
 
-  @spec notify_critical_database_health(binary()) :: :ok | {:error, any}
+  @spec notify_critical_database_health(binary()) :: :ok
   def notify_critical_database_health(id) do
     %DatabaseReadModel{sid: sid} = Trento.Repo.get!(DatabaseReadModel, id)
 
     EmailAlert.alert("Database", "SID", sid, "health is now in critical state")
     |> deliver_notification()
   end
 
-  @spec notify_critical_sap_system_health(binary()) :: :ok | {:error, any}
+  @spec notify_critical_sap_system_health(binary()) :: :ok
   def notify_critical_sap_system_health(id) do
     %SapSystemReadModel{sid: sid} = Trento.Repo.get!(SapSystemReadModel, id)
 
     EmailAlert.alert("Sap System", "SID", sid, "health is now in critical state")
     |> deliver_notification()
   end
 
+  @spec deliver_notification(Swoosh.Email.t()) :: :ok
   defp deliver_notification(%Swoosh.Email{subject: subject} = notification) do
     notification
```

Comment:
What do you think of leveraging `ENABLE_ALERTING` and skip delivery completely?

`alerting_enabled? = Application.fetch_env!(:trento, :alerting)[:enabled]`

and only try delivery when `true`